### PR TITLE
SLING-11468 : Simplify JcrPropertyMapCacheEntry.convertToType

### PR DESCRIPTION
**Pull request by anchela**
@cziegeler , @joerghoh , i would appreciate if you could take a look at the refactoring. in particular i tried to get rid of the duplicate conversion for strings and the fall-through behavior for conversion from inputstream that i found super hard to read.

### CODE REVIEW

**Review by joerghoh — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/sling/jcr/resource/internal/helper/JcrPropertyMapCacheEntry.java [hunk 1]
```diff
@@ -284,79 +284,84 @@ public boolean isArray() {
...
+        try (InputStream in = value) {
+            final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            final byte[] buffer = new byte[2048];
+            int l;
+            while ((l = in.read(buffer)) >= 0) {
+                if (l > 0) {
+                    baos.write(buffer, 0, l);
                 }
             }
+            return new String(baos.toByteArray(), StandardCharsets.UTF_8);
```
[the remark this answers is not in the document]
Comment (anchela) [reply src/main/java/org/apache/sling/jcr/resource/internal/helper/JcrPropertyMapCacheEntry.java:356]: i don't want to make major changes to the logic in a simple refactoring PR. can we cover this with a separate ticket?

File: src/test/java/org/apache/sling/jcr/resource/internal/helper/JcrPropertyMapCacheEntryTest.java [hunk 2]
```diff
@@ -119,6 +127,39 @@ public void testCharArray() throws Exception {
         verifyZeroInteractions(node);
     }
     
+    @Test
+    public void testCannotStore() throws Exception {
+        Object value = new TestClass();
+        try {
+            new JcrPropertyMapCacheEntry(value, node);
+            fail("IllegalArgumentException expected");
+        } catch (IllegalArgumentException e) {
```
[the remark this answers is not in the document]
Comment (anchela) [reply src/test/java/org/apache/sling/jcr/resource/internal/helper/JcrPropertyMapCacheEntryTest.java:136]: ma certo....

**Comment by anchela** [comment]
@joerghoh the potential memory issue with ByteArrayOutputStream should go into a separate bug report. the other findings i fixed. thanks a lot for taking a close look. let me know if there are other things left.

**Comment by anchela** [comment]
@joerghoh , can you take a look at my changes again an (hopefully) approve?

**Review by joerghoh — APPROVED** [review]

**Pull request merged**